Switch Git LFS install

mrjackv opened the pull request.

mrjackv committed:
install-system-dependencies: change lfs install

Change the way git LFS is installed from the `packagecloud.io` apt
repository to installing the release tarball from the github releases.

mrjackv committed:
ci/Dockerfile: fix broken COPY command

The script install-node-dependencies.sh was dropped a while ago, but a
reference remained in the `Dockerfile` for building the CI image. Drop
its use there as well.

mrjackv committed:
ci scripts: add `set -uo pipefail`

In addition to `set -e` also add `set -uo pipefail` to enforce stricter
execution of shell scripts.

aleclearmind commented on .orchestra/ci/install-system-dependencies.sh:
```diff
@@ -103,7 +103,7 @@ PACKAGES+=(ripgrep)
 apt-get -qq install --no-install-recommends --yes "${PACKAGES[@]}"
 
-if ! which git-lfs &> /dev/null; then
+if ! command -v git-lfs &> /dev/null; then
```
Why?

mrjackv replied:
`command -v` is:
* POSIX compatible
* a builtin of bash

Also there are some other (minor) subtle things about `which`, see [here](https://stackoverflow.com/a/677212)

aleclearmind replied:
ACK. IMO none of the arguments are particularly compelling here, but it's "better" in several ways.
Please adapt all the other uses of `which` in orchestra (or if too troublesome, leave `which`).

mrjackv replied:
Dropped all uses of `which`

aleclearmind commented on .orchestra/ci/install-dependencies.sh:
```diff
@@ -1,8 +1,8 @@
+set -euo pipefail
 
-SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+SCRIPT_DIR=$(realpath "$(dirname "${BASH_SOURCE[0]}")")
```
We use the old one everywhere. If you have a reason to change, you have to change it everywhere.

mrjackv replied:
Dropped

aleclearmind commented on .orchestra/ci/install-system-dependencies.sh:
```diff
@@ -104,6 +104,14 @@ PACKAGES+=(ripgrep)
+  wget --quiet "$LFS_URL" -O "$LFS_TMP/$LFS_FILE"
+  tar -C "$LFS_TMP" --strip-components=1 -xf "$LFS_TMP/$LFS_FILE"
+  cp -a "$LFS_TMP/git-lfs" /usr/local/bin
+  git lfs install --skip-repo --system &> /dev/null
```
I'd:
1. `curl -o- | tar xaf - ` so we avoid the temporary file
2. Extract only the file we need
3. Pin a git-lfs version and check the hash

mrjackv replied:
Ok, do note that this is different from the previous behavior where the latest version was fetched every time

mrjackv replied:
Done

mrjackv committed:
Drop `which`

The which utility is not standard POSIX and spawn a new process on each
invocation. Switch to `command` and `type` as they are bash builtins.

aleclearmind commented:
Merged in a1cc4c4. Thanks!
